### PR TITLE
Maintenance week updates

### PR DESCRIPTION
**Pull request by jonavellecuerdo**
### Purpose and background context

Update app according to [DataEng Application Maintenance](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/3445948428/DataEng+Application+Maintenance). Here are some changes worth noting:

* [Moved some sections that were previously in the README into `docs/`](https://github.com/MITLibraries/wiley-deposits/commit/555b73748d181c5993417bddfd17d0cd55215ac3).
* [Update `pyproject.toml` settings to use updated syntax](https://github.com/MITLibraries/wiley-deposits/commit/555b73748d181c5993417bddfd17d0cd55215ac3)
* Updated unit tests to use `mock_aws`
* Refactored `config` module to the clean, simplified version used in our more recent applications

### How can a reviewer manually see the effects of these changes?

1. Run `make lint`.
2. Run `make test`.

**Note**: Running the application in AWS is a little tricky as the full workflow depends on (1) a CSV from Wiley and (2) a DynamoDB table that keeps track of processing attempts. @ehanson8 and I agreed that running unit tests is sufficient for maintenance week testing. 

### Includes new or updated dependencies?
YES

### Changes expectations for external applications?
NO

### Developer
- [X] All new ENV is documented in README
- [X] All new ENV has been added to staging and production environments
- [X] All related Jira tickets are linked in commit message(s)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [x] The commit message is clear and follows our guidelines (not just this PR message)
- [x] There are appropriate tests covering any new functionality
- [x] The provided documentation is sufficient for understanding any new functionality introduced
- [x] Any manual tests have been performed **or** provided examples verified
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,143 +1,102 @@
...
-        alt DOI's process_attempts >= retry threshold 
-            wiley-app ->> dynamodb: Set DOI's status to 'FAILED'
-        else DOI's process_attempts < retry threshold
-            wiley-app ->> dynamodb: Reset DOI's status to 'UNPROCESSED'
-        end
-    end
-    wiley-app ->> ses: Send an email to stakeholders (subject: "DSS results")
+## Environment Variables
 
+### Required
```
Comment (ghukill) [inline README.md:27]: I like the inclusion and formatting of env vars in the main README.  Not a request for this PR, but maybe a topic for maintenance weeks could be settling on a formatting approach for all repositories when it comes to env vars.  I think we slowly coalescing around this code block, but could probably further refine a bit.

**Review by ghukill — APPROVED** [review]
Overall, think it looks great!  I like the movement of the sequence diagrams into standalone `docs` files.

I don't think any changes are needed to env vars formatting in the README, or `Config` class, but I see these two things as one of the most varying aspects of our projects still.   What's really helpful about these maintenance weeks is seeing them get _closer_ across repos, and perhaps soon we can discuss if there is an approach we'd like to aim for across all repos.  Said another way: glad to see them coalescing, no changes requested at this time, and think we have an opporunity to look at them again with fresh eyes soon now that they are getting more similar as possible maintenance week tweaks.

File: README.md (hunk 1 above)
Comment (jonavellecuerdo) [reply README.md:27]: Yeah. I think coming to a decision on the format of the "Environment Variables" section is a good idea (as well as capitalization rules for section headers). Both small things but I do think consistency/uniformity helps README navigation. For this README, I followed what you'd been doing (e.g. in `geo-harvester` but added a newline between each variable).

File: docs/metadata_mapping.md
Comment (ehanson8) [inline docs/metadata_mapping.md:1]: Good call on moving these, a much better place for them to live!

**Review by ehanson8 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ghukill) [reply README.md:27]: Yeah, hilariously, it's the newline -- again, don't think we should change it here! -- that kind of threw me.  I've come to expect env var sections (basically code) to be a little more compact and terse.  But it's ultimately just an agreed upon style, and looking forward to a style guide of sorts (even if just codified in our maintenance week updates flow).

**Pull request merged**